src/cpp/rtps/builtin/discovery/participant/PDPSimple.cpp

```diff
@@ -337,7 +336,6 @@ bool PDPSimple::createPDPEndpoints()
         delete mp_PDPWriterHistory;
         mp_PDPWriterHistory = nullptr;
         writer_payload_pool_->release_history(writer_pool_cfg, false);
-        TopicPayloadPoolRegistry::release(writer_payload_pool_);
```

Comment:
Shouldn't we do a `writer_payload_pool_.reset()` here?

Reply:
certainly!